getting_started/scripting/gdscript/gdscript_documentation_comments.rst

```diff
@@ -0,0 +1,219 @@
+with the double hash symbol ``##`` to be considered as it's part of the documentation.
+The script documentation will update in the editor help window every time the
+script updated and if any member variable or function name starts with an
+underscore and undocumented it'll treated as private for the documentation and
+ignored in the in the help window.
+
+Members that are applicable for the documentation
+
+- Inner class
+- Constant
+- Function
+- Signal
+- Variable
+- Enum
+- Enum value
+
+Examples
+--------
+
+::
+
+    extends Node2D
+	
+    ## 
+    ## A brief description about your script.
+    ##
+    ## @desc:
+    ##     The description about the script, what it
+    ##     can do, how to do that and everything.
+    ## 
+    ## @tutorial:            http://the/tutorial1/url.com
+    ## @tutorial(Tutorial2): http://the/tutorial2/url.com
+    ##
+    
+    ## the description about the variable v1
+    var v1
+    
+    ## multi line description about the variable v2. the type
+    ## information will be extracted below for the documentation
+    var v2: int
+    
+    ## if there are any annotation the member has, then it should
+    ## immediately precede them.
+    @export
+    @onready
+    var v3 := some_func()
+    
+    ## the description of a constant
+    const GRAVITY = 9.8
+	
+    ## the description of a signal
+    signal my_signal
+    
+    ## it's the enum description and see below the exception for 
+    ## the enum values which are documented on the same line.
+    enum Direction {
+        UP    = 0,  ## direction up
+        DOWN  = 1,  ## direction down
+        LEFT  = 2,  ## direction left
+        RIGHT = 3,  ## direction right
+    }
+    
+    ## since the below function is documented even it's name starts with
+    ## an underscore, it'll be updated in the editor.
+    func _fn(p1:int, p2:String) -> int:
+        return 0
+    
+    # the below function isn't documented and starts with an underscore name
+    # so this will treated as private and will not be updated in the editor.
+    func _internal() -> void:
+        pass
+    
+    ## About the inner class.
+    ## 
+    ## @desc: same rules like script description applies here, and It must
+    ##        immediately precede the class definition.
+    ##     
+    ## @tutorial: http://the/tutorial/url.com
+    class Inner:
+    
+        ## inner class variable v4
+        var v4
+    
+        ## inner class function fn 
+        func fn(): pass
+
+
+BBCode and class reference
+--------------------------
+
+The editor help window which renders the documentation supports :ref:`bbcode <doc_bbcode_in_richtextlabel>`,
+so it's possible to align and format the documentations and in addition color texts, images, fonts, tables,
+urls, animation effects, etc. could be added with the :ref:`bbcode <doc_bbcode_in_richtextlabel>`.
+
+Godot's class reference supports BBCode-like tags. They add nice formatting to the text which could also
```

Comment:
BBCode is already documented. Please change this to link to the existing BBcode documentation so it's not duplicated.